scrape/scrape.go

```diff
@@ -617,6 +617,47 @@ func (sp *scrapePool) refreshTargetLimitErr() error {
 	return nil
 }
 
+func (sp *scrapePool) schemesFromConfig() (validationScheme model.ValidationScheme, escapingScheme model.EscapingScheme, err error) {
+	// This is redundant to the config validation function but is easy to check anyway.
+	if model.NameValidationScheme == model.LegacyValidation && sp.config.MetricNameValidationScheme == config.UTF8ValidationConfig {
+		return model.LegacyValidation, model.UnderscoreEscaping, fmt.Errorf("cannot override library model.NameValidationScheme in config with allow-utf-8")
+	}
+
+	defaultValidation := model.NameValidationScheme
```

Comment:
Prometheus moved to UTF-8 now. I would say let's start with a UTF-8 default everywhere and gradually stop using the problematic global